Dockerfile.tmpl

```diff
@@ -25,6 +25,8 @@ ARG CUDA_MAJOR_VERSION
 ARG CUDA_MINOR_VERSION
 ENV CUDA_MAJOR_VERSION=${CUDA_MAJOR_VERSION}
 ENV CUDA_MINOR_VERSION=${CUDA_MINOR_VERSION}
+# Make sure we are on the right version of CUDA
+RUN update-alternatives --set cuda /usr/local/cuda-$CUDA_MAJOR_VERSION.$CUDA_MINOR_VERSION
```

Comment:
Is this because the latest DLVM image includes 2 versions of CUDA? Do we expect this command to go away once the base image is fixed? If so, should add a TODO with a link to a relevant bug stating that we should remove this.

Reply:
I don't particularly see any harm in having us explicitly select the version going forward.
Main advantages: Detect if the version we expect is/isn't installed right away so we can look into that (rather than be surprised like I was when multiple versions slipped in).

If the version is already correct, then this is basically a noop anyway.

Reply:
To clarify, yes this was introduced because the base image accidentally includes a partial install of CUDA 12.1
However I think its a worthwhile include regardless for future protection.